src/dapp/walletProvider.ts

```diff
@@ -1,240 +1,97 @@
-                    return;
-                }
-
-                const { data } = ev;
-                if (data.type !== DAPP_MESSAGE_CONNECT_URL) {
-                    return;
-                }
-
-                clearTimeout(timeout);
-                window.removeEventListener('message', connectUrl.bind(this));
-                return resolve(data.data.toString());
-            };
-
-            window.addEventListener('message', connectUrl);
-        }).then((connectionUrl: string) => {
-            let callbackUrl = `callbackUrl=${window.location.href}`;
-            if (options && options.callbackUrl) {
-                callbackUrl = `callbackUrl=${options.callbackUrl}`;
-            }
-            let token = '';
-            if (options && options.token) {
-                token = `&token=${options.token}`;
-            }
-            window.location.href = `${this.baseWalletUrl()}${connectionUrl}?${callbackUrl}${token}`;
-            return window.location.href;
-        }).catch(_ => {
-            return '';
-        });
+        window.location.href = `${this.baseWalletUrl()}${WALLET_PROVIDER_CONNECT_URL}?${callbackUrl}${token}`;
+        return window.location.href;
     }
 
     /**
     * Fetches the logout hook url and redirects the client to the wallet logout.
     */
-    async logout(): Promise<boolean> {
-        if (!this.mainFrame) {
-            return false;
+    async logout(options?: { callbackUrl?: string }): Promise<boolean> {
+        let callbackUrl = `callbackUrl=${window.location.href}`;
+        if (options && options.callbackUrl) {
+            callbackUrl = `callbackUrl=${options.callbackUrl}`;
         }
 
-        const { contentWindow } = this.mainFrame;
-        if (!contentWindow) {
-            console.warn("something went wrong, main wallet iframe does not contain a contentWindow");
-            return false;
-        }
-
-        return new Promise<boolean>((resolve, reject) => {
-            console.log("postMessage", DAPP_MESSAGE_LOG_OUT);
-
-            contentWindow.postMessage({
-                type: DAPP_MESSAGE_LOG_OUT,
-            }, this.walletUrl);
-
-            const timeout = setTimeout(_ => reject('logout url not responding'), 5000);
-            const logout = (ev: IDappMessageEvent) => {
-                console.log("event", "logouturl", ev);
-
-                if (!this.isValidWalletSource(ev.origin)) {
-                    return;
-                }
-
-                const { data } = ev;
-                if (data.type !== DAPP_MESSAGE_LOG_OUT) {
-                    return;
-                }
-
-                clearTimeout(timeout);
-                window.removeEventListener('message', logout.bind(this));
-                return resolve(data.data);
-            };
-
-            window.addEventListener('message', logout);
-        });
+        window.location.href = `${this.baseWalletUrl()}${WALLET_PROVIDER_DISCONNECT_URL}?${callbackUrl}`;
```

Comment:
can you please extract a function that composes these URLs and make unit tests for them?

Reply:
Did not extract into a function since it's a one liner, but managed to unit tests redirect urls (the trick here was to be able to mock `window.location.href`)